src/main/java/org/trigon/hardware/rev/spark/SparkMotor.java

```diff
@@ -12,111 +12,247 @@
+     *
+     * @return the ID
+     */
     public int getID() {
         return id;
     }
 
+    /**
+     * Registers a signal to be read from the motor.
+     *
+     * @param signal the signal to be read
+     */
     public void registerSignal(SparkSignal signal) {
         this.registerSignal(signal, false);
     }
 
+    /**
+     * Registers a signal to be read from the motor.
+     *
+     * @param signal     the signal to be read
+     * @param isThreaded whether the signal should be read in a separate thread
+     */
     public void registerSignal(SparkSignal signal, boolean isThreaded) {
         final SparkStatusSignal statusSignal = signal.getStatusSignal(motorIO.getMotor(), motorIO.getEncoder());
-        if (isThreaded)
+        if (isThreaded) {
             motorInputs.registerThreadedSignal(statusSignal);
-        else
-            motorInputs.registerSignal(statusSignal);
+            return;
+        }
+        motorInputs.registerSignal(statusSignal);
     }
 
+    /**
+     * Gets a signal from the motor.
+     *
+     * @param signal the signal to get
+     * @return the signal
+     */
     public double getSignal(SparkSignal signal) {
         return motorInputs.getSignal(signal.name);
     }
 
-    public void getThreadedSignal(SparkSignal signal) {
-        motorInputs.getThreadedSignal(signal.name);
+    /**
+     * Gets a threaded signal from the motor.
+     *
+     * @param signal the signal to get
+     * @return the signal
+     */
+    public double[] getThreadedSignal(SparkSignal signal) {
+        return motorInputs.getThreadedSignal(signal.name);
     }
 
-    public void setReference(double value, CANSparkBase.ControlType ctrl) {
-        motorIO.setReference(value, ctrl);
+    /**
+     * Sends a request to the motor.
+     *
+     * @param value       the value to set
+     * @param controlType the control type
+     */
+    public void setReference(double value, CANSparkBase.ControlType controlType) {
+        motorIO.setReference(value, controlType);
     }
 
-    public void setReference(double value, CANSparkBase.ControlType ctrl, int pidSlot) {
-        motorIO.setReference(value, ctrl, pidSlot);
+    /**
+     * Sends a request to the motor.
+     *
+     * @param value       the value to set
+     * @param controlType the control type
+     * @param pidSlot     the PID slot to use
+     */
+    public void setReference(double value, CANSparkBase.ControlType controlType, int pidSlot) {
+        motorIO.setReference(value, controlType, pidSlot);
     }
 
-    public void setReference(double value, CANSparkBase.ControlType ctrl, int pidSlot, double arbFeedForward) {
-        motorIO.setReference(value, ctrl, pidSlot, arbFeedForward);
+    /**
+     * Sends a request to the motor.
+     *
+     * @param value          the value to set
+     * @param controlType    the control type
+     * @param pidSlot        the PID slot to use
+     * @param arbFeedForward the feed forward value
```

Comment:
@levyishai is this meant to be `armFeedForward`?

Reply:
Check `motorIO` and find out

Reply:
What does motorIO have to do with it? It's `arbFeedForward` in the REV classes too and I assume that's why Yishai did it that way. But that doesn't mean it's not a mistake on REVs part.

Reply:
So check the REV code and maybe open a post on CD

Reply:
I checked the REV code, it's like that there too